documentation/testing_plans.md

```diff
@@ -0,0 +1,815 @@
+console together with each test's pass or fail status.
+
+## Configuration
+
+By default, the testing environment uses Bolt's default configuration and an
+empty inventory. The one exception is that Bolt's modulepath is configured to
+the modulepath set up for `rspec-puppet`.
+
+You can create your own values for Bolt's configuration, inventory, and
+modulepath by overriding the helper functions that set them. You can write
+these functions in the `spec_helper.rb` file, which exposes them to all tests,
+or in individual test files.
+
+### Overriding configuration
+
+To override the default configuration used by `BoltSpec`, write a `config_data`
+function in either your `spec_helper.rb` file or in the individual test files.
+This function should return a hash of configuration data that matches the
+structure expected in a [configuration file](bolt_project_reference.md).
+
+For example, the following `config_data` function loads configuration data from
+a test fixture:
+
+```ruby
+def config_data
+  YAML.load_file(File.expand_path('../fixtures/config.yaml'))
+end
+```
+
+### Overriding inventory
+
+To override the default inventory used by `BoltSpec`, write an `inventory_data`
+function. This function should return a hash of inventory data that matches the
+structure expected in an [inventory file](inventory_files.md).
+
+For example, the following `inventory_data` function loads inventory data from a
+test fixture:
+
+```ruby
+def inventory_data
+  YAML.load_file(File.expand_path('../fixtures/inventory.yaml'))
+end
+```
+
+### Overriding the modulepath
+
+To override the default modulepath used by `BoltSpec`, write a `modulepath`
+function. This function should return an array of strings, where each string is
+the path to a directory that includes modules.
+
+```ruby
+def modulepath
+  [File.expand_path('../fixtures/modules')]
+end
+```
+
+## Running plans
+
+The `BoltSpec` library includes the `run_plan` function which is used to run a
+plan. It has two required parameters: the name of the plan and a set of
+parameters to pass to the plan.
+
+To run a plan in a test, call the `run_plan` function:
+
+```ruby
+it 'calls the plan' do
+  run_plan('configure', { 'cleanup' => false })
+end
+```
+
+The `run_plan` function returns a [`PlanResult`
+object](bolt_types_reference.md#planresult), which you can examine and make
+assertions on. For example, if you expect your plan to succeed and return a
+message, you can write a test that makes those assertions.
+
+Given the following plan which returns a message:
+
+```puppet
+plan whoami () {
+  return 'Who are YOU?'
+}
+```
+
+You can write a test that asserts that the plan will return a successful result
+with the message `Who are YOU?`:
+
+```ruby
+it 'asks who you are' do
+  result = run_plan('whoami', {})
+
+  expect(result.ok?).to be(true)
+  expect(result.value).to eq('Who are YOU?')
+end
+```
+
+## Stubs and mocks
+
+The `BoltSpec` library includes several functions used to mock and stub executor
```

Comment:
```suggestion
The `BoltSpec` library includes several functions used to mock and stub plan
```

Reply:
While these go through the executor I think users are used to interacting with them through plans.